client/van_serviceinterface_update_test.go

```diff
@@ -0,0 +1,394 @@
+				trans,
+			},
+		},
+		{
+			doc:           "test three",
+			expectedError: "",
+			name:          "nginx",
+			port:          0,
+			eventChannel:  true,
+			aggregate:     "",
+			opts: []cmp.Option{
+				trans,
+			},
+		},
+		{
+			doc:           "test four",
+			expectedError: "Only one of aggregate and event-channel can be specified for a given service.",
+			name:          "nginx",
+			port:          0,
+			eventChannel:  true,
+			aggregate:     "json",
+			opts: []cmp.Option{
+				trans,
+			},
+		},
+		{
+			doc:           "test five",
+			expectedError: "invalidstrategy is not a valid aggregation strategy. Choose 'json' or 'multipart'.",
+			name:          "nginx",
+			port:          0,
+			eventChannel:  false,
+			aggregate:     "invalidstrategy",
+			opts: []cmp.Option{
+				trans,
+			},
+		},
+	}
+
+	var namespace string = "van-serviceinterface-update"
+	var cli *VanClient
+	var err error
+	if *clusterRun {
+		cli, err = NewClient(namespace, "", "")
+	} else {
+		cli, err = newMockClient(namespace, "", "")
+	}
+	assert.Assert(t, err)
+
+	_, err = kube.NewNamespace(namespace, cli.KubeClient)
+	defer kube.DeleteNamespace(namespace, cli.KubeClient)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	svcsFound := []string{}
+	svcsExpected := []string{"skupper-messaging", "skupper-internal", "skupper-controller", "nginx", "tcp-go-echo", "tcp-go-echo-ss"}
+
+	informers := informers.NewSharedInformerFactoryWithOptions(cli.KubeClient, 0, informers.WithNamespace(namespace))
+	svcInformer := informers.Core().V1().Services().Informer()
+	svcInformer.AddEventHandler(&cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			svc := obj.(*corev1.Service)
+			svcsFound = append(svcsFound, svc.Name)
+		},
+	})
+
+	informers.Start(ctx.Done())
+	cache.WaitForCacheSync(ctx.Done(), svcInformer.HasSynced)
+
+	// create three service targets
+	deployments := cli.KubeClient.AppsV1().Deployments(namespace)
+	statefulSets := cli.KubeClient.AppsV1().StatefulSets(namespace)
+
+	_, err = deployments.Create(tcpDeployment)
+	assert.Assert(t, err)
+	_, err = statefulSets.Create(tcpStatefulSet)
+	assert.Assert(t, err)
+	_, err = deployments.Create(httpDeployment)
+	assert.Assert(t, err)
+
+	err = cli.VanServiceInterfaceCreate(ctx, &types.ServiceInterface{
+		Address:      "noaddress",
+		Protocol:     "tcp",
+		Port:         12345,
+		EventChannel: false,
+		Aggregate:    "",
+	})
+	assert.Error(t, err, "Skupper not initialised in van-serviceinterface-update")
+
+	err = cli.VanRouterCreate(ctx, types.VanSiteConfig{
+		Spec: types.VanSiteConfigSpec{
+			SkupperName:       "skupper",
+			IsEdge:            false,
+			EnableController:  true,
+			EnableServiceSync: true,
+			EnableConsole:     false,
+			AuthMode:          "",
+			User:              "",
+			Password:          "",
+			ClusterLocal:      true,
+		},
+	})
+	assert.Assert(t, err)
+
+	// create three service definitions
+	time.Sleep(time.Second * 1)
```

Comment:
why is this sleep need?

Reply:
I think, we may run this same tests with clusters others that a cluster in localhost, probably we should at least unify this sleeps as a global configurable parameter, so we can reduce or increase then modifying a single line in the code... 
probably this comment is not only for this pr, and we can create an issue for that